### PR TITLE
Flow TasksConfiguration in build & specify generator config

### LINKED ISSUE #84157: BinClashes for Microsoft.Interop.SourceGeneration.dll
### Description

I opened a binlog and noticed this:
image

That indicates the build is building both debug and release copies of Microsoft.Interop.SourceGeneration.dll.  We end up shipping whichever one builds last.

### Reproduction Steps

Build the product and examine binlog.

### Expected behavior

No binclashes

### Actual behavior

Binclashes for Microsoft.Interop.SourceGeneration.dll

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/runtime/issues/84157

Builds on top of the previous state of https://github.com/dotnet/runtime/pull/84666

1. Flow the TasksConfiguration in the root build (Subsets.props)
2. Specify the LibrariesConfiguration explicitly when building the
   LibraryImport source generators.
3. Set tasks configuration to debug by default.

@elinor-fung I worked on that mid last week and hadn't found time to push. I think this is a slightly better approach as it uses the common path (Debug/Release config for tasks) and makes sure that the TasksConfiguration and the LibrariesConfiguration is available in when triggered from a root build. This builds on top of your existing work which looked really good.

### CODE REVIEW

**Comment by ViktorHofer** [comment]
@elinor-fung did you had a chance to look at this PR?

File: src/tasks/tasks.proj [hunk 1]
```diff
@@ -24,7 +24,7 @@
     </ItemGroup>
 
     <MSBuild Projects="@(TaskProject)"
-             Properties="Configuration=Debug;Platform=AnyCPU"
+             Properties="Configuration=$(TasksConfiguration);Platform=AnyCPU"
```
Comment (elinor-fung) [inline src/tasks/tasks.proj:27]: The ProjectToBuild.AdditionalProperties from Subset.props won't be flowed the the individual task projects here, right? Should this also explicitly propagate `LibrariesConfiguration=$(LibrariesConfiguration)`?

Comment (elinor-fung) [reply src/tasks/tasks.proj:27]: @ViktorHofer I think this is my only question here.
Everything else looks good.

Comment (ViktorHofer) [reply src/tasks/tasks.proj:27]: Oh wow, I completely dropped the ball here. Let me rebase on top of main, look at the binlog and then think about your question. Thanks for the ping.

**Comment by elinor-fung** [comment]
@ViktorHofer did you have a chance to get back to this? Since it fixes binclashes that could affect shipping bits, we'd like to get it fixed for 8.

I can also take a look if you are heads down on other things.

**Comment by ViktorHofer** [comment]
I'm out this week but can take a look next week. Sorry for the delay. I will make sure that this is ready before we branch off.

**Comment by ViktorHofer** [comment]
@elinor-fung the PR is now ready. We still have two double writes but those are unrelated and I filed https://github.com/dotnet/runtime/issues/89763 for it.

**Review by elinor-fung — APPROVED** [review]
Thank you for doing this!

**Pull request merged**